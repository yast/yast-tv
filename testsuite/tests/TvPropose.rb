# encoding: utf-8

module Yast
  class TvProposeClient < Client
    def main
      # testedfiles: Tv.ycp Sound.ycp sound/write_routines.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      @R = {}
      @W = {}
      @E = { "audio" => { "alsa" => { "store" => true } } }

      # Tv::cards == nil
      Tv.cards = nil
      Tv.detected_cards = [
        {
          "module"     => ["bttv"],
          "name"       => "Second detected",
          "unique_key" => "123xyz"
        }
      ]
      TEST(lambda { Tv.Propose }, [@R, @W, @E], nil)
      DUMP(Tv.cards)

      # Tv::cards == card1
      Tv.cards = [
        {
          "module"        => ["bttv"],
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        }
      ]
      Tv.detected_cards = [] # detected_cards cannot have nil value
      TEST(lambda { Tv.Propose }, [@R, @W, @E], nil)
      DUMP(Tv.cards)

      # Tv::cards == card1, nil, nil, card4
      Tv.cards = [
        {
          "module"        => ["bttv"],
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        {},
        {},
        {
          "module"        => ["xyz"],
          "parameters"    => { "par_other" => "testvalue2" },
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      Tv.detected_cards = [
        {
          "module"     => ["zoran"],
          "name"       => "First detected",
          "unique_key" => "klic"
        },
        {
          "module"     => ["bttv"],
          "name"       => "Second detected",
          "unique_key" => "123xyz"
        }
      ]
      TEST(lambda { Tv.Propose }, [@R, @W, @E], nil)
      DUMP(Tv.cards)

      nil
    end
  end
end

Yast::TvProposeClient.new.main
