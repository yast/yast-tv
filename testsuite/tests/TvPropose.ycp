{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	// Tv::cards == nil
	Tv::cards = nil;
	Tv::detected_cards = [ $["module":"bttv", "name":"Second detected", "unique_key":"123xyz"]];
	TEST(``(Tv::Propose()), [], []);
	DUMP(Tv::cards);

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	Tv::detected_cards = []; // detected_cards cannot have nil value
	TEST(``(Tv::Propose()), [], []);
	DUMP(Tv::cards);

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	Tv::detected_cards = [ $["module":"zoran", "name":"First detected", "unique_key":"klic"],
						   $["module":"bttv", "name":"Second detected", "unique_key":"123xyz"]];
	TEST(``(Tv::Propose()), [], []);
	DUMP(Tv::cards);
}
