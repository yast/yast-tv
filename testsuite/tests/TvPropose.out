Dir	.audio.alsa.cards: []
Execute	.audio.alsa.store 0 0 true
Return	nil
Dump	[$["module":"bttv", "name":"Second detected", "unique_key":"123xyz"]]
Return	nil
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"]]
Dir	.audio.alsa.cards: []
Execute	.audio.alsa.store 0 0 true
Return	nil
Dump	[$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"], $["module":"zoran", "name":"First detected", "unique_key":"klic"], $["module":"bttv", "name":"Second detected", "unique_key":"123xyz"], $["module":"xyz", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]]
